src/AzureStorageQueueWorker.cs

```diff
@@ -63,20 +63,31 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
             using var logScope = _logger.BeginScope("{Processor} reads {queueName}", processorName, _workerOptions.QueueName);
             try
             {
+                await Task.Yield();
                 var queueClient = new QueueClient(_workerOptions.AzureStorageConnectionString, _workerOptions.QueueName);
                 await queueClient.CreateIfNotExistsAsync();
                 while (!stoppingToken.IsCancellationRequested)
                 {
                     var processed = false;
-                    MessageDetails messageDetails = null;
-                    QueueMessage queueMessage = null;
-                    CancellationTokenSource updateMessageTokenSource = null;
-                    Task updateQueueMessageTask = null;
-                    IServiceScope scope = null;
+                    MessageDetails? messageDetails = null;
+                    QueueMessage? queueMessage = null;
+                    CancellationTokenSource? updateMessageTokenSource = null;
+                    Task? updateQueueMessageTask = null;
+                    IServiceScope? scope = null;
+                    IOperationHolder<DependencyTelemetry>? operation = null;
                     try
                     {
-                        var queueMessageResponse = await queueClient.ReceiveMessagesAsync(1, _workerOptions.VisibilityTimeout, stoppingToken);
-                        queueMessage = queueMessageResponse.Value.FirstOrDefault();
+                        //clean values here before any code
+                        //otherwisewe can get previous loop cycle value in the catch/finally
+                        queueMessage = null;
+                        messageDetails = null;
+                        updateQueueMessageTask = null;
+                        updateMessageTokenSource = null;
```

Comment:
Can you elaborate on this? - I don't understand how getting the previous loop value is possible since these variables are all defined at the beginning of the loop

Reply:
if the following operation fails on the second loop
https://github.com/Likvido/Likvido.Worker.AzureStorageQueue/blob/dbaf813691823e9baa1153e91b4ab55bc7f1fff1/src/AzureStorageQueueWorker.cs#L87-L88

Reply:
The loop starts on line 69, and all of those variables are defined and initialized on lines 72 - 75 ... so they would be null - even on second loop. 

There must be a detail I am missing?

Reply:
yeah, you are right :)
I'll fix this shortly